src/chronify/time_series_checker.py

```diff
@@ -1,79 +1,79 @@
-            for col in time_columns:
-                stmt = stmt.where(table.c[col].is_not(None))
-            df = read_database(stmt, conn, schema)
-            actual = set(schema.time_config.list_timestamps_from_dataframe(df))
-            match = sorted(actual) == expected
-            # TODO: This check doesn't work and I'm not sure why.
-            # diff = actual.symmetric_difference(expected)
-            # if diff:
-            #     msg = f"Actual timestamps do not match expected timestamps: {diff}"
-            #     # TODO: list diff on each side.
-            #     raise InvalidTable(msg)
-            if not match:
-                msg = "Actual timestamps do not match expected timestamps"
-                # TODO: list diff on each side.
-                raise InvalidTable(msg)
+    def _check_expected_timestamps(self) -> None:
+        expected = self._schema.time_config.list_timestamps()
+        time_columns = self._schema.time_config.list_time_columns()
+        stmt = select(*(self._table.c[x] for x in time_columns)).distinct()
+        for col in time_columns:
+            stmt = stmt.where(self._table.c[col].is_not(None))
+        df = read_database(stmt, self._conn, self._schema)
+        actual = self._schema.time_config.list_distinct_timestamps_from_dataframe(df)
+        match = actual == expected
+        # TODO: This check doesn't work and I'm not sure why.
+        # diff = actual.symmetric_difference(expected)
+        # if diff:
+        #     msg = f"Actual timestamps do not match expected timestamps: {diff}"
+        #     # TODO: list diff on each side.
+        #     raise InvalidTable(msg)
+        if not match:
+            msg = "Actual timestamps do not match expected timestamps"
+            # TODO: list diff on each side.
+            raise InvalidTable(msg)
 
-    def _check_expected_timestamps_by_time_array(self, schema: TableSchema) -> None:
-        with self._engine.connect() as conn:
-            tmp_name = make_temp_view_name()
-            self._run_timestamp_checks_on_tmp_table(schema, conn, tmp_name)
-            conn.execute(text(f"DROP TABLE IF EXISTS {tmp_name}"))
+    def _check_expected_timestamps_by_time_array(self) -> None:
+        tmp_name = make_temp_view_name()
+        self._run_timestamp_checks_on_tmp_table(tmp_name)
+        self._conn.execute(text(f"DROP TABLE IF EXISTS {tmp_name}"))
 
-    @staticmethod
-    def _run_timestamp_checks_on_tmp_table(
-        schema: TableSchema, conn: Connection, table_name: str
-    ) -> None:
-        id_cols = ",".join(schema.time_array_id_columns)
-        filters = [f"{x} IS NOT NULL" for x in schema.time_config.list_time_columns()]
+    def _run_timestamp_checks_on_tmp_table(self, table_name: str) -> None:
+        id_cols = ",".join(self._schema.time_array_id_columns)
+        filters = [f"{x} IS NOT NULL" for x in self._schema.time_config.list_time_columns()]
         where_clause = "AND ".join(filters)
```

Comment:
This needs to be " AND ", space in front of and behind.

Reply:
Thanks. Fixed.